File: 2540_minimum_common_value.php
<?php

class Solution {

  /**
  * @param int[] $nums1
  * @param int[] $nums2
  * @return int
  */
  function getCommon($nums1, $nums2) {

    foreach ($nums1 as $num1) {
      $startIndexOfNum2 = 0;
      for ($i = $startIndexOfNum2; $i < count($nums2); $i++, $startIndexOfNum2++) {
        if ($num1 == $nums2[$i]) {
          return $num1;
        }

        if ($num1 < $nums2[$i]) {
          break;
        }
      }

    }

    return -1;
  }
}

$solution = new Solution();

$nums1 = [1, 3, 5, 7, 9];
$nums2 = [2, 4, 6, 8, 10];

echo $solution->getCommon($nums1, $nums2); // -1

$nums1 = [1, 3, 5, 7, 9];
$nums2 = [2, 4, 6, 8, 9];

echo $solution->getCommon($nums1, $nums2); // 9

File: 2540_minimum_common_value.java
class Solution {
    public int getCommon(int[] nums1, int[] nums2) {

      for (int i = 0; i < nums1.length; i++) {
        int startIndexOfNums2 = 0;
        for (int j = startIndexOfNums2; j < nums2.length; j++, startIndexOfNums2++) {
          if (nums1[i] == nums2[j]) {
            return nums1[i];
          }

          if (nums1[i] < nums2[j]) {
            break;
          }
        }
      }

      return -1;
    }
}
